### ISSUE TITLE
Allow `/` in collection names

### ISSUE THREAD

**Issue by w84thesun** [FerretDB/FerretDB#1777]
Probably should be done in or after #2210.

---

### What should be done?

We should allow `/` in collection names because a lot of MongoDB Go driver tests depend on it: https://github.com/FerretDB/dance/blob/f3ad4b770066c5794550b629ef5cb46ed2c6e09f/tests/mongo-go-driver.yml#L104-L110

We should do it for PostgreSQL. We should also try to do it for Tigris. One way to do it is to add a new parameter to `testutil.CollectionName` to generate different names.

### Where?

* https://github.com/FerretDB/FerretDB/blob/b996cff36d630a277f236c6dfc45cbe198089566/internal/util/testutil/db.go#L92
* https://github.com/FerretDB/FerretDB/blob/b996cff36d630a277f236c6dfc45cbe198089566/internal/handlers/pg/pgdb/collections.go#L34-L35

### Definition of Done

- integration tests added;
- compatibility tests added;
- diff tests updated;
- spot refactorings done;
- user documentation updated.

**Comment by b1ron** [comment]
We should probably allow certain non-latin characters in collection names as well. For example a lot of the Go driver tests fail because collection names are generated from sub-tests which contain `/`:

```
    --- FAIL: TestCollection/find_one_and_delete (0.05s)
        --- FAIL: TestCollection/find_one_and_delete/found (0.01s)
            mongotest.go:473: error creating collection or view: TestCollection/find_one_and_delete/found on server: (InvalidNamespace) Invalid collection name: 'test.TestCollection/find_one_and_delete/found'
```

**Comment by kropidlowsky** [comment]
I would like to work on it

**Comment by AlekSi** [comment]
Fixed by #2210.

```
test> db.getCollection('/').insertOne({})
{
  acknowledged: true,
  insertedId: ObjectId("641c22afd9589005095eb1f0")
}
test> db.getCollection('/').find()
[ { _id: ObjectId("641c22afd9589005095eb1f0") } ]
```
